pkg/api/validation/validation.go

```diff
@@ -1032,6 +1033,25 @@ func ValidateService(service *api.Service) errs.ValidationErrorList {
 		}
 	}
 
+	// Check for duplicate NodePorts
+	nodePorts := []int{}
+	for i := range service.Spec.Ports {
+		if service.Spec.Ports[i].NodePort != 0 {
+			nodePorts = append(nodePorts, service.Spec.Ports[i].NodePort)
```

Comment:
You need to accumulate and validate {protocol, port} pairs.  We have this bug elsewhere, too.